projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_error_classifier.py

```diff
@@ -0,0 +1,135 @@
+        or _is_redshift_error(received_exception)
+        or _is_other_query_error(received_exception)
+        or _is_property_unsupported_value_error(received_exception)
+    )
+
+
+def _is_syntax_error(exception):
+    return (
+        errors.exception_matches(
+            exception,
+            classname_with_package="impala.error.HiveServer2Error",
+            exception_message_matcher_regex="^AnalysisException.*",
+        )
+        or errors.exception_matches(
+            exception,
+            classname_with_package="impala.error.HiveServer2Error",
+            exception_message_matcher_regex="^ParseException: Syntax error.*",
+        )
+        or errors.exception_matches(
+            exception,
+            classname_with_package="impala.error.HiveServer2Error",
+            exception_message_matcher_regex="^Could not compile regexp pattern.*",
+        )
+        or errors.exception_matches(
+            exception,
+            classname_with_package="impala.error.HiveServer2Error",
+            exception_message_matcher_regex="^TableNotFoundException.*",
+        )
+        or errors.exception_matches(
+            exception,
+            classname_with_package="impala.error.HiveServer2Error",
+            exception_message_matcher_regex=".*ArrayIndexOutOfBoundsException:.*",
+        )
+    )
+
+
+def _is_udf_error(exception):
+    return errors.exception_matches(
+        exception,
+        classname_with_package="impala.error.OperationalError",
+        exception_message_matcher_regex="^UDF ERROR:.*",
+    )
+
+
+def _is_authorization_error(exception):
+    return errors.exception_matches(
+        exception,
+        classname_with_package="impala.error.HiveServer2Error",
+        exception_message_matcher_regex="^AuthorizationException:.*",
+    )
+
+
+def _is_query_limit_exceeded(exception):
+    return (
+        # Error specified below occurs when query exceeds 1h time limit
+        errors.exception_matches(
+            exception,
+            classname_with_package="impala.error.OperationalError",
+            exception_message_matcher_regex="^Cancelled from Impala's debug web interface$",
+        )
+        or errors.exception_matches(
+            exception,
+            classname_with_package="impala.error.OperationalError",
+            exception_message_matcher_regex="^.*expired due to execution time limit.*$",
+        )
+    )
+
+
+def _is_other_query_error(exception):
+    return errors.exception_matches(
+        exception,
+        classname_with_package="impala.error.OperationalError",
+        exception_message_matcher_regex=".*Cannot perform hash join at node with id.*",
+    )
+
+
+def _is_property_unsupported_value_error(exception):
+    return errors.exception_matches(
+        exception,
+        classname_with_package="ValueError",
+        exception_message_matcher_regex=".*[pP]roperty.*is of unsupported type.*",
+    )
+
+
+def _is_impala_memory_limit_exceeded(exception):
```

Comment:
Looking at memory limit exceeded conditions, I am missing 
```
(errors.exception_matches(
        exception,
        classname_with_package="MemoryError",
        exception_message_matcher_regex=".*memory limit.*")
```

Also, could combine with the common `_is_query_memory_error()` one into one memory-related block

Reply:
```
(errors.exception_matches(
        exception,
        classname_with_package="MemoryError",
        exception_message_matcher_regex=".*memory limit.*")
```
This is not an impala error. It is a generic python memory error and it is covered by the core job input error classifier: https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/builtin_plugins/run/job_input_error_classifier.py#L72